tests_e2e/tests/recover_network_interface/recover_network_interface.py

```diff
@@ -0,0 +1,124 @@
+# limitations under the License.
+#
+
+#
+# This test uses CSE to bring the network down and call check_and_recover_nic_state to bring the network back into an
+# 'up' and 'connected' state. The intention of the test is to alert us if there is some change in newer distros which
+# affects this logic.
+#
+
+import json
+
+from assertpy import fail, assert_that
+from time import sleep
+
+from tests_e2e.tests.lib.agent_test import AgentVmTest
+from tests_e2e.tests.lib.agent_test_context import AgentVmTestContext
+from tests_e2e.tests.lib.logging import log
+from tests_e2e.tests.lib.virtual_machine_extension_client import VirtualMachineExtensionClient
+from tests_e2e.tests.lib.vm_extension_identifier import VmExtensionIds
+
+
+class RecoverNetworkInterface(AgentVmTest):
+    def __init__(self, context: AgentVmTestContext):
+        super().__init__(context)
+        self._context = context
+        self._ssh_client = context.create_ssh_client()
+        self._private_ip = context.vm.get_private_ip_address()
+        self._vm_password = ""
+
+    def add_vm_password(self):
+        # Add password to VM to help with debugging in case of failure
+        # REMOVE PWD FROM LOGS IF WE EVER MAKE THESE RUNS/LOGS PUBLIC
+        username = self._ssh_client.username
+        pwd = self._ssh_client.run_command("openssl rand -base64 32 | tr : .").rstrip()
+        self._vm_password = pwd
+        log.info("VM Username: {0}; VM Password: {1}".format(username, pwd))
+        self._ssh_client.run_command("echo '{0}:{1}' | sudo -S chpasswd".format(username, pwd))
+
+    def check_agent_reports_status(self):
+        status_updated = False
+        last_agent_status_time = self._context.vm.get_instance_view().vm_agent.statuses[0].time
+        log.info("Agent reported status at {0}".format(last_agent_status_time))
+        retries = 3
+
+        while retries > 0 and not status_updated:
+            agent_status_time = self._context.vm.get_instance_view().vm_agent.statuses[0].time
+            if agent_status_time != last_agent_status_time:
+                status_updated = True
+                log.info("Agent reported status at {0}".format(last_agent_status_time))
+            else:
+                retries -= 1
+                sleep(60)
+
+        if not status_updated:
+            fail("Agent hasn't reported status since {0} and ssh connection failed. Use the serial console in portal "
+                 "to debug".format(last_agent_status_time))
+
+    def run(self):
+        # Add password to VM and log. This allows us to debug with serial console if necessary
+        log.info("")
+        log.info("Adding password to the VM to use for debugging in case necessary...")
+        self.add_vm_password()
+
+        # Get the primary network interface name
+        ifname = self._ssh_client.run_command("pypy3 -c 'from azurelinuxagent.common.osutil.redhat import RedhatOSUtil; print(RedhatOSUtil().get_if_name())'").rstrip()
```

Comment:
there is no need to start another instance of pypy3, you could simple import the Agent's module from this test.